src/query/interpret/eval.hpp

```diff
@@ -714,6 +752,9 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
         result.emplace("microsecond", TypedValue(lt.microsecond, ctx_->memory));
         return TypedValue(result, ctx_->memory);
       }
+      case TypedValue::Type::ZonedDateTime: {
+        throw QueryRuntimeException("Can't coerce `{}` to Map.", expression_result.ValueZonedDateTime().ToString());
```

Comment:
Either way is doable; in this case I chose this way for the following reasons:
* It’s in line with other graph databases.
* The year...microsecond components aren’t actual members (whereas they are in e.g. `LocalDateTime`); they’re rather computed properties
* More consistent for users - If in the future we add new ZonedDateTime components (e.g. week, epochMicroseconds or offset), existing queries with AllPropertyLookup (i.e. map projections) will change since the map projection should contain them as new fields). By forbidding ZonedDateTime in AllPropertyLookup, this is avoided.